packages/components/src/tab-panel/index.tsx

```diff
@@ -97,11 +97,13 @@ export function TabPanel( {
 	const selectedId = `${ instanceId }-${ selectedTab?.name ?? 'none' }`;
 
 	useEffect( () => {
-		const newSelectedTab = find( tabs, { name: selected } );
-		if ( ! newSelectedTab && tabs.length > 0 ) {
-			handleTabSelection( initialTabName || tabs[ 0 ].name );
```

Comment:
Careful on the updated logic here, apparently it was just merged a few days ago from #44028. I proposed some added tests in #45211.